### ISSUE TITLE
zprint -e exception

### ISSUE THREAD

**Issue by agriffis** [kkinnear/zprint#116]
This seems unexpected...

```
✸ zprintl-0.5.1 -e
Exception in thread "main" java.util.concurrent.ExecutionException: java.lang.IllegalArgumentException: No matching field found: isArray for class java.lang.Class
        at java.util.concurrent.FutureTask.report(FutureTask.java:122)
        at java.util.concurrent.FutureTask.get(FutureTask.java:192)
        at clojure.core$deref_future.invokeStatic(core.clj:2292)
        at clojure.core$future_call$reify__8097.deref(core.clj:6894)
        at clojure.core$deref.invokeStatic(core.clj:2312)
        at clojure.core$pmap$step__8110$fn__8114.invoke(core.clj:6945)
        at clojure.lang.LazySeq.sval(LazySeq.java:40)
        at clojure.lang.LazySeq.seq(LazySeq.java:49)
        at clojure.lang.RT.seq(RT.java:528)
        at clojure.core$seq__5124.invokeStatic(core.clj:137)
        at clojure.core$filter$fn__5614.invoke(core.clj:2801)
        at clojure.lang.LazySeq.sval(LazySeq.java:40)
        at clojure.lang.LazySeq.seq(LazySeq.java:49)
        at clojure.lang.RT.seq(RT.java:528)
        at clojure.lang.RT.countFrom(RT.java:643)
        at clojure.lang.RT.count(RT.java:636)
        at zprint.zprint$contains_nil_QMARK_.invokeStatic(zprint.cljc:277)
        at zprint.zprint$contains_nil_QMARK_.invoke(zprint.cljc:273)
        at zprint.zprint$fzprint_map_two_up.invokeStatic(zprint.cljc:1090)
        at zprint.zprint$fzprint_map_two_up.invoke(zprint.cljc:1029)
        at zprint.zprint$fzprint_map_STAR_.invokeStatic(zprint.cljc:4028)
        at zprint.zprint$fzprint_map_STAR_.invoke(zprint.cljc:3936)
        at zprint.zprint$fzprint_map.invokeStatic(zprint.cljc:4079)
        at zprint.zprint$fzprint_map.invoke(zprint.cljc:4057)
        at zprint.zprint$fzprint_STAR_.invokeStatic(zprint.cljc:4517)
        at zprint.zprint$fzprint_STAR_.invoke(zprint.cljc:4461)
        at zprint.zprint$fzprint.invokeStatic(zprint.cljc:5104)
        at zprint.zprint$fzprint.invoke(zprint.cljc:5092)
        at clojure.core$partial$fn__5565.invoke(core.clj:2629)
        at zprint.sutil$sredef_call$fn__4365.invoke(sutil.cljc:353)
        at zprint.redef$redef_vars.invokeStatic(redef.cljc:98)
        at zprint.redef$redef_vars.invoke(redef.cljc:61)
        at zprint.sutil$sredef_call.invokeStatic(sutil.cljc:280)
        at zprint.sutil$sredef_call.invoke(sutil.cljc:276)
        at zprint.core$fzprint_style.invokeStatic(core.cljc:203)
        at zprint.core$fzprint_style.invoke(core.cljc:150)
        at zprint.core$zprint_STAR_.invokeStatic(core.cljc:322)
        at zprint.core$zprint_STAR_.invoke(core.cljc:302)
        at zprint.core$zprint_str_internal.invokeStatic(core.cljc:387)
        at zprint.core$zprint_str_internal.doInvoke(core.cljc:366)
        at clojure.lang.RestFn.invoke(RestFn.java:425)
        at clojure.lang.AFn.applyToHelper(AFn.java:156)
        at clojure.lang.RestFn.applyTo(RestFn.java:132)
        at clojure.core$apply.invokeStatic(core.clj:661)
        at clojure.core$apply.invoke(core.clj:652)
        at zprint.core$zprint_str.invokeStatic(core.cljc:475)
        at zprint.core$zprint_str.doInvoke(core.cljc:459)
        at clojure.lang.RestFn.invoke(RestFn.java:410)
        at zprint.main$_main.invokeStatic(main.cljc:118)
        at zprint.main$_main.doInvoke(main.cljc:62)
        at clojure.lang.RestFn.applyTo(RestFn.java:137)
        at zprint.main.main(Unknown Source)
Caused by: java.lang.IllegalArgumentException: No matching field found: isArray for class java.lang.Class
        at clojure.lang.Reflector.getInstanceField(Reflector.java:271)
        at clojure.lang.Reflector.invokeNoArgInstanceMember(Reflector.java:315)
        at zprint.sutil$sarray_QMARK_.invokeStatic(sutil.cljc:176)
        at zprint.sutil$sarray_QMARK_.invoke(sutil.cljc:172)
        at zprint.zprint$fzprint_STAR_.invokeStatic(zprint.cljc:4521)
        at zprint.zprint$fzprint_STAR_.invoke(zprint.cljc:4461)
        at zprint.zprint$fzprint_two_up.invokeStatic(zprint.cljc:780)
        at zprint.zprint$fzprint_two_up.invoke(zprint.cljc:700)
        at clojure.lang.AFn.applyToHelper(AFn.java:186)
        at clojure.lang.AFn.applyTo(AFn.java:144)
        at clojure.core$apply.invokeStatic(core.clj:663)
        at clojure.core$partial$fn__5567.doInvoke(core.clj:2635)
        at clojure.lang.RestFn.invoke(RestFn.java:408)
        at clojure.core$pmap$fn__8105$fn__8106.invoke(core.clj:6942)
        at clojure.core$binding_conveyor_fn$fn__5476.invoke(core.clj:2022)
        at clojure.lang.AFn.call(AFn.java:18)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
        at com.oracle.svm.core.thread.JavaThreads.threadStartRoutine(JavaThreads.java:460)
        at com.oracle.svm.core.posix.thread.PosixJavaThreads.pthreadStartRoutine(PosixJavaThreads.java:193)
```

**Comment by kkinnear** [comment]
Thanks for catching this.  Interestingly, it works fine with the uberjar, but it turns out that `-e` has not worked with the graalVM binaries for quite some time (if ever).  If you really need to know this information, the uberjar (`java -jar zprint-filter-0.5.1 -e`) should be able to give you the same information successfully.  It takes a bit of time to start, so it isn't necessarily a good choice for actually doing the formatting, but it will give you output for `-e`.  As well as I can tell, the graalVM binaries format successfully and correctly.  I'll be working on figuring out what is going on with the `-e` output.

**Comment by kkinnear** [comment]
Fixed in 0.5.2.  It was solved by adding a type hint so that graalVM could figure out how to resolve a method call.  Speedy technology, but lots of hidden issues along the way.  Thanks for noticing this problem.

**Comment by agriffis** [comment]
Thanks!

**Comment by agriffis** [comment]
@kkinnear I'm not sure if you're keeping this open for some reason, but it's fixed in 0.5.3 in my testing. Thanks!

**Comment by kkinnear** [comment]
Thanks, no, I just didn't want to close it if you weren't done with it.
